bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/v1/ModelServerClientV1.java

```diff
@@ -44,7 +44,7 @@
 
    protected static final Logger LOG = LogManager.getLogger(ModelServerClientV1.class);
 
-   private final ModelServerClientDelegate delegate;
+   protected final ModelServerClientDelegate delegate;
```

Comment:
This (together with making the delegate class public which would be required) is the only change that gives me pause. The idea of the delegate was that it was intended only to provide common behaviour of the core routing classes. Subclasses outside of the model server framework should not need to access the delegate directly, but only via APIs (preferably protected) of the V1/V2 classes. If necessary, those APIs should be added to the V1/V2 classes.

What specifically motivates opening this up to the world at large?

Reply:
Thanks for your feedback on that @cdamus !
Oh I understand, yes this sounds of course very reasonable!

My specific use case is an extension of the `ModelServerClientV1` atm, which adds a custom request. It would therefore be great to reuse the `createHttpUrlBuilder` oder the `makeCallAndGetDataBody` methods (which are also restricted atm).
My code would look currently like this:

```
   @SuppressWarnings("restriction")
   public CompletableFuture<Response<List<String>>> getTypes(final String modelUri) {
      final Request request = new Request.Builder()
         .url(
            // createHttpUrlBuilder(delegate.getBaseUrl() + MyPaths.TYPES)
            delegate.createHttpUrlBuilder(baseUrl + MyPaths.TYPES)
               .addQueryParameter(ModelServerPathParametersV1.MODEL_URI, modelUri)
               .build())
         .build();

      return this.delegate.makeCallAndGetDataBody(request)
         .thenApply(response -> response.mapBody(body -> {
            List<String> uris = new ArrayList<>();
            try {
               Json.parse(body).forEach(uri -> uris.add(uri.textValue()));
               return uris;
            } catch (IOException e) {
               throw new CompletionException(e);
            }
         }));
   }
```
Should I extract this as a util method perhaps?